static/js/util/util.js

```diff
@@ -50,6 +51,38 @@ export function makeCourseStatusDisplay(course, now = moment()) {
     courseOrRun = course.runs[0];
   }
 
+  let onEnrollBtnClick = (event) => {
+    event.preventDefault();
+    let url = [];
+    if (courseOrRun.course_id) {
+      url.push(
+        SETTINGS.edx_base_url,
+        'courses/',
+        courseOrRun.course_id,
+        '/info'
+      );
+      window.open(url.join(""), '_blank');
+    } else {
+      alert("Please enter valid course id");
```

Comment:
A course run will always have a course id (course key). If there is no course run, there will be no course id and we shouldn't attempt to display an enroll or upgrade button.